.github/workflows/pr.yaml

```diff
@@ -41,6 +43,40 @@ jobs:
     uses: rapidsai/shared-workflows/.github/workflows/checks.yaml@branch-24.10
     with:
       enable_check_generated_files: false
+  fetch-pandas-versions:
+    runs-on: ubuntu-latest
+    outputs:
+      pandas-versions: ${{ steps.get-versions.outputs.versions }}
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v2
+
+      - name: Set up Python
+        uses: actions/setup-python@v2
+        with:
+          python-version: '3.11'
+
+      - name: Install requests
+        run: pip install requests
+      - name: Install packaging
+        run: pip install packaging
+      - name: Fetch pandas versions
+        id: get-versions
+        run: |
+          versions=$(python ci/cudf_pandas_scripts/fetch_pandas_versions.py ">=2.0,<2.2.3dev0")
+          echo "::set-output name=versions::${versions}"
+  cross-pandas-version-tests:
+    needs: [fetch-pandas-versions, wheel-build-cudf]
+    secrets: inherit
+    strategy:
+      matrix:
+        pandas-version: ${{ fromJson(needs.fetch-pandas-versions.outputs.pandas-versions) }}
```

Comment:
How many jobs is this? Three? What exact versions? 2.0.0, 2.1.0, 2.2.0, or is it testing every bugfix release too? How long does each job take to execute?

We need to be cautious about the GPU CI runtime here on every PR.

Reply:
It will test every minor version `2.0`, `2.1`, `2.2`

Reply:
How long does each job take?

Depending on time, since every job in the matrix is going to use the same Python+CUDA combination, it might make sense to run the tests of in sequence in the script on a single GPU job instead of allocating 3 GPU machines.

Reply:
Using this PR for example (https://github.com/rapidsai/cudf/actions/runs/10481802900/job/29032506344?pr=16623), looks like almost 5 minutes to run pytest over `cudf_pandas_tests`. So IMO it should be OK to run these tests in serial for each pandas minor version

Reply:
Yes, it takes 4 mins for job setup and 4 mins for jobs to run. So I felt it's right to just squeeze all this into one job. Pushed the changes.